src/main/scala/scala/tools/partest/PartestDefaults.scala

```diff
@@ -25,7 +25,9 @@ object PartestDefaults {
     import scala.reflect.io.Path, Path._
     Some(Path(jdkHome) / "bin") filter (_.isDirectory) flatMap { p =>
       val candidates = (p walkFilter { e => (e.name == name || e.name.startsWith(s"$name.")) && e.jfile.canExecute }).toList
-      (candidates find (_.name == name) orElse candidates.headOption) map (_.path)
+      val plain = candidates find (_.name == name)
+      val exe = candidates find (_.name == s"${name}.exe")
+      (plain orElse exe orElse candidates.headOption).map(_.path)
```

Comment:
Your second expr gets it done. With paranoid case insensitive.
```
(Path(jdkHome) / "bin").walk.find(e => e.name == name || isWin && e.name.equalsIgnoreCase(s"$name.exe")).map(_.path)
```
When I looked yesterday, the doc on walkFilter (there's doc) inverts the sense of filter, so I wouldn't assume anything about experimental i/o library. But walk on a regular file seems OK.